lib/Transforms/ElementwiseToAffine/ElementwiseToAffine.cpp

```diff
@@ -70,7 +70,11 @@ struct ConvertAnyElementwiseMappableOpOnRankedTensors : public RewritePattern {
 
       // If first loop: replace scalar op
       if (i == 0) {
-        rewriter.replaceOp(op, loop);
+        // We will cast back to the return type to retain an embedding if
+        // included
+        auto cast = rewriter.create<tensor::CastOp>(op->getLoc(), resultType,
+                                                    loop.getResults());
```

Comment:
Nice, yes there is! Changed to set the type directly in the `tensor.empty` op and removed the changes to the elementwise testcases.

Reply:
A nice improvement. I did check up if this would have been handled by canonicalization: `FoldEmptyTensorWithCastOp` is upstream, but it requires the encoding attributes to be the same to apply the folder.

Reply:
> Nice, yes there is! Changed to set the type directly in the `tensor.empty` op and removed the changes to the elementwise testcases.

Awesome, thanks!